### PR TITLE
Fixes AirTerminal Mixer used with FanCoilUnit does not respond to fan availability schedule

### LINKED ISSUE #8107: Inlet AT Mixer used with Fan Coil does not respond to fan availability schedule
Issue overview
--------------
User file shows AT Mixer operating when fan coil is off. The reason is that the fan coil is not considering the fan availability schedule before simulating the AT mixer. The left figure shows the fan coil and AT Mixer flow rates at night when the night cycle manager is turning the system on and off. The right (and left) figure shows the AT Mixer flow rate is non-zero during this time. The problem shows on Jan 2 of the weather file simulation (La Crosse Municipal Arpt,WI).

image

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added (EnergyPlusDevSupport 8107 defect folder)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by Nigusse**
Pull request overview
---------------------
 - Fixes #8107
 - Now the FancoilUnit responds to NightCycle availability manager. 
 - The defect file, which has a night cycle availability manager, behaves as expected, i.e., the FanCoil Unit operation matches the availability status. 

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [x] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [x] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [x] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by Nigusse** [comment]
These two plots show the defect and the fix. The fix shows that the fancoilunit cycles consistently with the status of the nighcycle availability manager.

ATMixerFlow_CyclingStatus_befreandAfter

**Comment by rraustad** [comment]
This looks correct now and is the same result back when this was fixed in Trane work last year.

File: src/EnergyPlus/FanCoilUnits.cc [hunk 1]
```diff
@@ -1308,7 +1315,7 @@ namespace FanCoilUnits {
             }
         }
         // Set the inlet node mass flow rate
-        if ((GetCurrentScheduleValue(FanCoil(FanCoilNum).SchedPtr) > 0.0 || DataHVACGlobals::ZoneCompTurnFansOn) &&
+        if (((GetCurrentScheduleValue(FanCoil(FanCoilNum).SchedPtr) > 0.0 && GetCurrentScheduleValue(FanCoil(FanCoilNum).fanAvailSchIndex) > 0.0) || DataHVACGlobals::ZoneCompTurnFansOn) &&
```
Comment (rraustad) [inline src/EnergyPlus/FanCoilUnits.cc]: This line seems long, was this formatted by Clang?

File: src/EnergyPlus/FanCoilUnits.cc [hunk 2]
```diff
@@ -3538,7 +3545,7 @@ namespace FanCoilUnits {
 
         // Assume the unit is able to vary the flow. A cycling unit is treated as
         // if it were variable flow, with the flow being the averaqe flow over the time step
-        if ((GetCurrentScheduleValue(FanCoil(FanCoilNum).SchedPtr) > 0.0 || DataHVACGlobals::ZoneCompTurnFansOn) &&
+        if (((GetCurrentScheduleValue(FanCoil(FanCoilNum).SchedPtr) > 0.0 && GetCurrentScheduleValue(FanCoil(FanCoilNum).fanAvailSchIndex) > 0.0) || DataHVACGlobals::ZoneCompTurnFansOn) &&
```
Comment (rraustad) [inline src/EnergyPlus/FanCoilUnits.cc]: Clang formatted?

File: tst/EnergyPlus/unit/AirTerminalSingleDuctMixer.unit.cc [hunk 3]
```diff
@@ -7767,4 +7768,415 @@ TEST_F(EnergyPlusFixture, AirTerminalSingleDuctMixer_SimFCU_ATMInletSideTest)
...
+    // set predicted heating load
+    zoneSysEnergyDemand.RemainingOutputReqToCoolSP = 4000.0;
+    zoneSysEnergyDemand.RemainingOutputReqToHeatSP = 4000.0;
+    zoneSysEnergyDemand.RemainingOutputRequired = 4000.0;
+    QZnReq = zoneSysEnergyDemand.RemainingOutputRequired;
+    QUnitOut = 0.0;
+    QLatOut = 0.0;
+    InitFanCoilUnits(state, FanCoilNum, ZoneNum, ZoneNum);
+    Sim4PipeFanCoil(state, FanCoilNum, ZoneNum, ZoneNum, FirstHVACIteration, QUnitOut, QLatOut);
+    // check results when the fna coil unit is not available
```
Comment (rraustad) [inline tst/EnergyPlus/unit/AirTerminalSingleDuctMixer.unit.cc]: fna -> fan

File: src/EnergyPlus/FanCoilUnits.cc (hunk 1 above)
Comment (Nigusse) [reply src/EnergyPlus/FanCoilUnits.cc]: I did not try to change the format, it must be auto-formated by Clang.

**Review by rraustad — APPROVED** [review]
These changes are correct.

File: tst/EnergyPlus/unit/AirTerminalSingleDuctMixer.unit.cc (hunk 3 above)
Comment (Nigusse) [reply tst/EnergyPlus/unit/AirTerminalSingleDuctMixer.unit.cc]: Let the CI machine run complete and I will do formatting and cleanup.

File: src/EnergyPlus/FanCoilUnits.cc (hunk 1 above)
Comment (rraustad) [reply src/EnergyPlus/FanCoilUnits.cc]: If you select/highlight this line and then on the Edit tab choose Advanced and select "Format Selection" clang should format the line. Do the same for line 3548.

**Comment by Nigusse** [comment]
Ok. I applied Clang formatting to the two locations and corrected the typo.

**Comment by Nigusse** [comment]
I see unrelated diffs and this diffs popped when I removed unused variable. May be it is lagging develop, so I will merge in develop and push.

**Comment by Myoldmopar** [comment]
Results look perfect and CI is happy.  I believe this is ready to go.  @rraustad merge away if you don't have anything further.

**Comment by rraustad** [comment]
Agreed. I didn't expect any diff's and the plots provided show expected behavior.

**Pull request merged**